### PR TITLE
style: `Homepage` design

### LINKED ISSUE #689: feat: redesign Homepage
Design a better Homepage which showcases:= `AnimeCore` `MangaCore` `ForumCore`

### PR DESCRIPTION
**Pull request by tokitouq**
Closes #689

### CODE REVIEW

**Comment by tokitou-san** [comment]
@baseplate-admin take a loook
Screenshot from 2023-09-09 21-17-43

**Comment by baseplate-admin** [comment]
https://github.com/tokitou-san/CoreProject-Frontend/blob/e7833b7966de0804fbb27224aa29bd733d31f973/src/routes/%2Bpage.svelte#L63


we want this in typewriter too

File: package.json [hunk 1]
```diff
@@ -86,6 +86,7 @@
         "tippy.js": "^6.3.7",
         "tslib": "^2.6.2",
         "typescript": "^5.2.2",
+        "typewriter-effect": "^2.20.1",
```
Comment (baseplate-admin) [inline package.json:89]: Are you serious? This is a react library :\

**Review by baseplate-admin — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (tokitou-san) [reply package.json:89]: Huh! we're using its core, yea it supports react btw ( same like floatingUI ) skele skele lol
```ts
import Typewriter from "typewriter-effect/dist/core";
```

Comment (baseplate-admin) [reply package.json:89]: > import Typewriter from "typewriter-effect/dist/core";

Yes no we want something better :\ . The repo is not updated in like months

**Comment by baseplate-admin** [comment]
> LOL okay? @baseplate-admin could you add this? 

Yes and no. I dont like this library at all. This one doesn't tell us when this thing is finished.

**Comment by tokitou-san** [comment]
> This one doesn't tell us when this thing is finished.

Indeed, but why you want this function? if you want to only type upper text once add `loop` false.
also for bottom things add `pauseFor` for certain time ( upper text finish time ). uhh lol I'm not sure

But I like how it looks now ( no typing on upper text )

File: src/routes/+page.svelte [hunk 2]
```diff
@@ -1,7 +1,44 @@
...
+                .typeString('<span class="bg-gradient-to-r from-primary-500 to-primary-200 bg-clip-text text-transparent">Anime. </span>')
+                .pauseFor(2000)
+                .deleteAll()
+                .typeString('<span class="bg-gradient-to-r from-primary-400 to-error-400 bg-clip-text text-transparent">Manga. </span>')
+                .pauseFor(2000)
+                .deleteAll()
+                .typeString('<span class="bg-gradient-to-r from-warning-400 to-error-300 bg-clip-text text-transparent">Sound. </span>')
+                .pauseFor(2000)
+                .start()
+            }, 2500)
```
Comment (baseplate-admin) [inline src/routes/+page.svelte:40]: see why i told you to get a library that tells you when it's finished?

File: src/routes/+page.svelte [hunk 3]
```diff
@@ -17,4 +54,45 @@
...
+            </a>
+        </div>
+    </navbar>
+    <landing class="flex items-center justify-between h-screen md:px-[10vw]">
+        <content>
+            <div class="flex flex-col md:gap-[2vw] relative">
+                <div class="flex flex-col md:text-[5vw] font-bold md:leading-[4.5vw]">
+                    <!-- <span>Unlock the World of</span> -->
+                    <main-typewriter bind:this={top_typewriter_el} />
+                    <typewriter bind:this={typewriter_el} />
```
Comment (baseplate-admin) [inline src/routes/+page.svelte:82]: Also make this `user-select:none`

Comment (tokitou-san) [reply src/routes/+page.svelte]: Okay I dont think this is good.
let them select
also now we're rendering `a` tags so noo :P

**Comment by baseplate-admin** [comment]
see the layout shift

image

File: src/routes/+page.svelte [hunk 4]
```diff
@@ -17,4 +37,56 @@
...
+                            class="btn rounded-[0.4rem] bg-gradient-to-r from-primary-500 to-error-500 px-5 py-4 text-base font-semibold leading-none md:rounded-[0.5vw] md:px-[1.25vw] md:py-[1vw] md:text-[1vw] shadow-2xl shadow-primary-800"
+                        >
+                            <span>Get Started</span>
+                            <Chevron class="md:w-[1.25vw] -rotate-90" />
+                        </button>
+                    </cta-buttons>
+                </div>
+            </div>
+        </content>
+        <mascot class="relative flex flex-shrink-0 md:h-[40vw] self-end">
```
Comment (baseplate-admin) [inline src/routes/+page.svelte:84]: Give this fixed height

File: src/routes/+page.svelte [hunk 5]
```diff
@@ -17,4 +37,56 @@
...
+            <img src="images/characters/hibiki/hibiki.png" alt="Hibiki" class="z-10 h-full" />
+            <!-- gradients -->
+            <gradient class="absolute left-1/4 md:w-[15vw] md:h-[20vw] rotate-45 bg-primary-900 md:rounded-[2vw] md:blur-[3.5vw]" />
+            <gradient class="absolute md:bottom-[2.5vw] right-0 md:w-[15vw] md:h-[20vw] rotate-45 bg-primary-800 md:rounded-[2vw] md:blur-[4vw]" />
+            <gradient class="absolute md:bottom-[1vw] left-1/4 md:w-[10vw] md:h-[20vw] rotate-45 bg-surface-400 md:rounded-[2vw] md:blur-[3.5vw]" />
```
Comment (baseplate-admin) [inline src/routes/+page.svelte:89]: Everything should be absolute

**Review by baseplate-admin — CHANGES_REQUESTED** [review]

File: src/routes/+page.svelte (hunk 5 above)
Comment (tokitou-san) [reply src/routes/+page.svelte:89]: Everything? that `img` too?

File: src/routes/+page.svelte (hunk 4 above)
Comment (tokitou-san) [reply src/routes/+page.svelte:84]: Uhh you mean fixed width?

File: src/routes/+page.svelte (hunk 5 above)
Comment (baseplate-admin) [reply src/routes/+page.svelte:89]: Yes

File: src/routes/+page.svelte (hunk 4 above)
Comment (baseplate-admin) [reply src/routes/+page.svelte:84]: Yep

File: src/routes/+page.svelte (hunk 5 above)
Comment (tokitou-san) [reply src/routes/+page.svelte]: Uhh I think we dont need this since we added `self-end`.
absolute isn't going to make any changes

**Pull request closed without merging**